async_producer_test.go

```diff
@@ -1695,12 +1715,16 @@ func TestAsyncProducerInterceptors(t *testing.T) {
 		},
 		{
 			name:         "interceptor chain",
-			interceptors: []ProducerInterceptor{&appendInterceptor{i: 0}, &appendInterceptor{i: 1000}},
+			interceptors: []ProducerInterceptor{&appendInterceptor{i: 0}, &appendInterceptor{i: 1000}, &metadataInterceptor{text: "sarama"}},
 			expectationFn: func(t *testing.T, i int, msg *ProducerMessage) {
 				v, _ := msg.Value.Encode()
-				expected := TestMessage + strconv.Itoa(i) + strconv.Itoa(i+1000)
-				if string(v) != expected {
-					t.Errorf("Interceptor should have incremented the value, got %s, expected %s", v, expected)
+				expectedValue := TestMessage + strconv.Itoa(i) + strconv.Itoa(i+1000)
+				if string(v) != expectedValue {
+					t.Errorf("Interceptor should have incremented the value, got %s, expected %s", v, expectedValue)
+				}
+
+				if str, ok := msg.Metadata.(string); ok != true || str != "sarama" {
```

Comment:
Do not compare `bool` values to true or false, just use `!ok`

Reply:
fixed, thanks